src/cleanlab_codex/validation.py

```diff
@@ -0,0 +1,190 @@
+"""
+This module provides validation functions for checking if an LLM response is unhelpful.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable, Optional
+
+from cleanlab_codex.utils.prompt import default_format_prompt
+
+if TYPE_CHECKING:
+    from cleanlab_studio.studio.trustworthy_language_model import TLM  # type: ignore
+
+
+DEFAULT_FALLBACK_ANSWER = "Based on the available information, I cannot provide a complete answer to this question."
+DEFAULT_PARTIAL_RATIO_THRESHOLD = 70
+DEFAULT_TRUSTWORTHINESS_THRESHOLD = 0.5
+
+
+def is_bad_response(
+    response: str,
+    context: str,
+    tlm: TLM,  # TODO: Make this optional
+    query: Optional[str] = None,
+    # is_fallback_response args
+    fallback_answer: str = DEFAULT_FALLBACK_ANSWER,
```

Comment:
```suggestion
def is_bad_response(
    response: str,
    *,
    query: Optional[str] = None,
    context: Optional[str] = None,
    tlm:  Optional[TLM] = None,
    # is_fallback_response args
    fallback_answer: Optional[str] = DEFAULT_FALLBACK_ANSWER,
```

Reply:
i vote to make lots of these optional, and the `is_bad_response` function internally just doesn't run a certain check if the input args required for that check were not provided.  Keep the code simple though, I wouldn't make it sophisticated.

Reply:
Some of these could just have default values instead of being `Optional`.
Or do you mean we should swap some of these with `None?
E.g.

```diff
-fallback_answer: str = DEFAULT_FALLBACK_ANSWER,
+fallback_answer: Optional[str] = None,
```

Reply:
no sorry my code is bad above, I just meant to suggest the user doesn't actually have to specify any of these (eg. `context`, `tlm`) and to re-order these arguments. otherwise you can disregard the specific code above.